### ISSUE TITLE
WHITESPACE rule triggers Clippy

### ISSUE THREAD

**Issue by HollayHorvath** [pest-parser/pest#490]
The `WHITESPACE` rule has a fixed name, but it triggers the `clippy::upper_case_acronyms` lint. Since this name gets exported to Rust in the `#[grammar]` macro, this rule cannot be silenced locally.
I think the `WHITESPACE` rule should be always silent (at least I cannot imagine a situation where one would like to use this rule as-is), so I think it should not be exported to Rust at all, but I'm not familiar with the internals.

**Comment by ian-h-chamberlain** [comment]
Looking at `cargo expand` output, it seems like there are already some existing lints disabled for the generated rules...
```rust
    #[allow(dead_code, non_camel_case_types)]
    pub enum Rule {
        WHITESPACE,
        Line,
        Statement,
        Expression,
        // ...
    }
```

So maybe it could just be added there in the `allow()` attribute as a quicker fix as opposed to keeping the `WHITESPACE` variant unexposed.

Meanwhile, wrapping the derive in a module is a workaround that allows me to disable the lint but avoid changing other code referencing the grammar or rules:
```rust
#[allow(clippy::upper_case_acronyms)]
mod derived {
    use pest_derive::Parser;

    #[derive(Parser)]
    #[grammar = "grammar/redcode.pest"]
    pub struct Grammar;
}

pub use derived::{Grammar, Rule};
```

**Comment by CAD97** [comment]
This should be fixed by #507.

**Comment by MarinPostma** [comment]
this is indeed fixed by #507
